detect_secrets/plugins/keyword.py

```diff
@@ -144,17 +172,20 @@
 }
 # Includes ], ', " as closing
 CLOSING = r'[]\'"]{0,2}'
-DENYLIST_REGEX = r'|'.join(DENYLIST)
+AFFIX_REGEX = r'[a-zA-Z0-9_-]*'
+DENYLIST_REGEX = r'(' + r'|'.join(DENYLIST) + r')({suffix})?'.format(suffix=AFFIX_REGEX)
+DENYLIST_REGEX_WITH_PREV = r'({prefix})('.format(prefix=AFFIX_REGEX) \
+    + r'|'.join(DENYLIST) + r')({suffix})'.format(suffix=AFFIX_REGEX)
 # Non-greedy match
 OPTIONAL_WHITESPACE = r'\s*?'
 OPTIONAL_NON_WHITESPACE = r'[^\s]{0,50}?'
 QUOTE = r'[\'"]'
-SECRET = r'[^\s]+'
+SECRET = r'""|\'\'|[^\'"\s]+'
```

Comment:
Why do we consider `""` and `''` a secret?

Reply:
When you analyze an other person's repository, you don't known the context of each secret, so I think that it is important to detect secrets in the mosts situations as possible and then mark them as false positive if it is needed. In some environments, as the local environment, developers usually configure empty passwords for some resources, so I think that detect-secrets should detect it.

Reply:
Moreover, there are a lot of common technologies that use blank passwords by default. [Here](https://github.com/danielmiessler/SecLists/blob/master/Passwords/Default-Credentials/default-passwords.csv) you can see some.